File: 0792-binary-search/solution.c
int search(int* arr, int n, int target) {
    int low=0;
    int high=n-1;
    int result=-1;
    while(low<=high){
        int mid=(low+high)/2;
        if(arr[mid]==target){
            result=mid;
            break;
        }
        else if(target>arr[mid]){
            low=mid+1;
        }
        else if(arr[mid]>target){
            high=mid-1;
        }
    }
    return result;
}

File: 0792-binary-search/solution.java
class Solution {
    public int search(int[] nums, int target) {
     int i=0;
     int j = nums.length-1;
     int mid=0;
     while(i<j){
        mid = i+(j-i)/2;
        if(nums[mid]==target){
            return mid;
        }
        else if(target<nums[mid]){
            j=mid;
        }
        else{
            i=mid+1;
        }
     }   
    //  mid=i+(j-i)/2;
     if(nums[i]==target){
     return i;}
     else{
        return -1;
     }
    }
}
